src/Common/src/TypeSystem/TypesDebugInfoWriter/UserDefinedTypeDescriptor.cs

```diff
@@ -0,0 +1,158 @@
+        }
+
+        public ulong  GetEnumRecordValue(FieldDesc field)
+        {
+            var ecmaField = field as EcmaField;
+            if (ecmaField != null)
+            {
+                MetadataReader reader = ecmaField.MetadataReader;
+                FieldDefinition fieldDef = reader.GetFieldDefinition(ecmaField.Handle);
+                ConstantHandle defaultValueHandle = fieldDef.GetDefaultValue();
+                if (!defaultValueHandle.IsNil)
+                {
+                    return HandleConstant(ecmaField.Module, defaultValueHandle);
+                }             
+            }
+            return 0;
+        }
+
+        private ulong HandleConstant(EcmaModule module, ConstantHandle constantHandle)
+        {
+            MetadataReader reader = module.MetadataReader;
+            Constant constant = reader.GetConstant(constantHandle);
+            BlobReader blob = reader.GetBlobReader(constant.Value);
+            switch (constant.TypeCode)
+            {
+                case ConstantTypeCode.Byte:
+                    return (ulong)blob.ReadByte();
+                case ConstantTypeCode.Int16:
+                    return (ulong)blob.ReadInt16();
+                case ConstantTypeCode.Int32:
+                    return (ulong)blob.ReadInt32();
+                case ConstantTypeCode.Int64:
+                    return (ulong)blob.ReadInt64();
+                case ConstantTypeCode.SByte:
+                    return (ulong)blob.ReadSByte();
+                case ConstantTypeCode.UInt16:
+                    return (ulong)blob.ReadUInt16();
+                case ConstantTypeCode.UInt32:
+                    return (ulong)blob.ReadUInt32();
+                case ConstantTypeCode.UInt64:
+                    return (ulong)blob.ReadUInt64();
+            }
+            System.Diagnostics.Debug.Assert(false);
+            return 0;
+        }
+
+        public uint GetClassTypeIndex(TypeDesc type)
+        {
+            DefType defType = type as DefType;
+            System.Diagnostics.Debug.Assert(defType != null, "GetClassTypeIndex was called with non def type");
+            ClassTypeDescriptor classTypeDescriptor = new ClassTypeDescriptor();
+            classTypeDescriptor.IsStruct = type.IsValueType? 1:0;
+            classTypeDescriptor.Name = defType.Name;
+            classTypeDescriptor.UniqueName = defType.GetFullName();
+            classTypeDescriptor.BaseClassId = 0;
+            if (type.HasBaseType && !type.IsValueType)
+            {
+                classTypeDescriptor.BaseClassId = _objectWriter.GetVariableTypeIndex(defType.BaseType);
+            }
+            uint result = _knownTypes[type] = _objectWriter.GetClassTypeIndex(classTypeDescriptor);
+
+            List <DataFieldDescriptor> fieldsDescs = new List<DataFieldDescriptor>();
+            foreach (var fieldDesc in defType.GetFields())
+            {
+                if (fieldDesc.HasRva || fieldDesc.IsLiteral)
```

Comment:
What fields are you interested in here? Should this actually filter out all of the static fields? How do we know in the debugger what base (instance?, GC static? threadstatic?) does the field refer to?